sql/mysql_db_structure.sql

```diff
@@ -399,6 +399,7 @@ CREATE TABLE `scorecache` (
   `pending_public` int(4) unsigned NOT NULL DEFAULT '0' COMMENT 'Number of submissions pending judgement (public)',
   `solvetime_public` decimal(32,9) NOT NULL DEFAULT '0.000000000' COMMENT 'Seconds into contest when problem solved (public)',
   `is_correct_public` tinyint(1) unsigned NOT NULL DEFAULT '0' COMMENT 'Has there been a correct submission? (public)',
+  `is_solved_first` tinyint(1) unsigned NOT NULL DEFAULT '0' COMMENT 'Is this the first solution to this problem?',
```

Comment:
There should be a `restricted` and `public` version of this, since a first solve may happen in the freeze.

Reply:
Why would you need restricted and public versions? If the fts happens after freeze, is_correct_public will be 0 so there's no problem?

Reply:
Ah, ok. Makes sense, so the use of `is_solved_first` is dependent on `is_correct_*`.